managers/backup-manager/BackupManager.js

```diff
@@ -0,0 +1,350 @@
+          operationType: CONST.APISERVER.ANNOTATION_TYPES.BACKUP,
+          operationId: result.backup_guid,
+          stateValue: 'in_progress'
+        }).then(() =>
+          eventmesh.server.updateOperationResult({
+            resourceId: opts.instance_guid,
+            operationName: CONST.APISERVER.ANNOTATION_NAMES.BACKUP,
+            operationType: CONST.APISERVER.ANNOTATION_TYPES.BACKUP,
+            operationId: result.backup_guid,
+            value: backup_options
+          })
+        )
+      )
+      .then(() => {
+        return eventmesh.server.getOperationResult({
+          resourceId: opts.instance_guid,
+          operationName: CONST.APISERVER.ANNOTATION_NAMES.BACKUP,
+          operationType: CONST.APISERVER.ANNOTATION_TYPES.BACKUP,
+          operationId: result.backup_guid,
+        });
+      })
+      .catch(err => {
+        return Promise
+          .try(() => logger.error(`Error during start of backup - backup to be aborted : ${backupStarted} - backup to be deleted: ${metaUpdated} `, err))
+          .tap(() => eventmesh.server.updateOperationState({
+            resourceId: opts.instance_guid,
+            operationName: CONST.APISERVER.ANNOTATION_NAMES.BACKUP,
+            operationType: CONST.APISERVER.ANNOTATION_TYPES.BACKUP,
+            operationId: result.backup_guid,
+            stateValue: CONST.APISERVER.RESOURCE_STATE.ERROR
+          }))
+          .tap(() => eventmesh.server.updateOperationResult({
+            resourceId: opts.instance_guid,
+            operationName: CONST.APISERVER.ANNOTATION_NAMES.BACKUP,
+            operationType: CONST.APISERVER.ANNOTATION_TYPES.BACKUP,
+            operationId: result.backup_guid,
+            value: err
+          }))
+          .tap(() => {
+            if (backupStarted) {
+              logger.error(`Error occurred during backup process. Aborting backup on deployment : ${deploymentName}`);
+              return this
+                .abortLastBackup(opts, true)
+                .finally(() => {
+                  if (metaUpdated) {
+                    const options = _
+                      .chain(data)
+                      .pick(data, 'tenant_id', 'backup_guid')
+                      .set('force', true)
+                      .value();
+                    logger.error(`Error occurred during backup process. Deleting backup file on deployment : ${deploymentName} - backup file:`, options);
+                    return this.backupStore
+                      .deleteBackupFile(options);
+                  }
+                })
+                .catch((err) => logger.error('Error occurred while performing clean up of backup failure operation : ', err));
+            }
+          }).then(() => {
+            throw err;
+          });
+      });
+  }
+
+  getServiceFabrikOperationState(name, opts) {
+    logger.info(`Retrieving state of last Backup with:`, opts);
+    return Promise
+      .try(() => {
+        switch (name) {
+        case 'backup':
```

Comment:
We do not need the switch case. This will anyways be only for Backup.

Reply:
Done!